src/core/posix.cc

```diff
@@ -122,16 +122,16 @@ posix_thread::posix_thread(attr a, std::function<void ()> func)
     }
 #endif
 
-    if (a._affinity) {
-        auto& cpuset = *a._affinity;
-        pthread_attr_setaffinity_np(&pa, sizeof(cpuset), &cpuset);
-    }
-
     r = pthread_create(&_pthread, &pa,
                 &posix_thread::start_routine, _func.get());
     if (r) {
         throw std::system_error(r, std::system_category());
     }
+
```

Comment:
might want to explain why we need to move this `if` clause down in the commit message. also, we have a tradition of adding prefix in the title of commit message, so the title should look like:

> posix: ...

and the title is not accurate, this commit does not move a function.

Reply:
I am not sure why, @neopsykose wrote it.  I will build without that patch and see what happens.

Reply:
I tried without that patch and the error is:

```
src/core/posix.cc:127:9: error: 'pthread_attr_setaffinity_np' was not declared in this scope; did you mean 'pthread_setaffinity_np'?
  127 |         pthread_attr_setaffinity_np(&pa, sizeof(cpuset), &cpuset);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~
      |         pthread_setaffinity_np
```

Reply:
I don’t think that one is musl specific?

Reply:
strictly speaking, none of the changes in this PR is. they are the consequence of the differences of glibc and non-glibc, which in this case, musl libc implementations.

in this very case, as the name of `pthread_attr_setaffinity_np` suggests, this API is not portable, but a GNU extension. without which, we cannot set the affinity when creating a pthread.

but without another extension `pthread_setaffinity_np`, we can set the CPU affinity **after** the thread is created. hence this change. since both glibc and musl libc implements this non-portable extension, we can use it here as an alternative of `pthread_attr_setaffinity_np`.

if you are with me. could you please put more context and the rationales in the corresponding commit message?